tests/helpers/app-fixture.js

```diff
@@ -0,0 +1,206 @@
+var fs = require('fs');
+var path = require('path');
+var stringUtil = require('ember-cli-string-utils');
+var execSync = require('child_process').execSync;
+var symlinkOrCopySync = require('symlink-or-copy').sync;
+
+var packageCache = require('./package-cache');
+
+var fixturify = require('fixturify');
+var quickTemp = require('quick-temp');
+var merge = require('ember-cli-lodash-subset').merge;
+
+function AppFixture(name, options) {
+  this.type = 'app';
+  this.name = name;
+  this.options = options || {
+    useGlobalPackages: true
+  };
+
+  this.fixture = {};
+  this._fixtureCache = {};
+  this.dirs = {};
+  quickTemp.makeOrRemake(this.dirs, 'self');
+
+  this.blueprintPath = path.resolve(__dirname, '../../blueprints/app');
+  this.blueprintOptionsShim = {
```

Comment:
I'm going to refactor this work on top of `blprnt`, this was a first pass to prove things out. I'm coming back to this tomorrow-ish.